gadopt/stokes_integrators.py

```diff
@@ -322,6 +334,9 @@ def set_solver_options(self) -> None:
         self.appctx = {
             "mu": self.approximation.mu / self.approximation.rho_continuity()
         }
+        if self.free_surface_map:
+            self.appctx["free_surface"] = self.free_surface_map
+            self.appctx["ds"] = self.equations[-1].ds
```

Comment:
Why did this move?

Reply:
True, maybe too early. When using an ABC, there is no reason to have this condition for all solvers when it is only used by `StokesSolver`. As a result, it makes more sense to have it in solver options than setup, as it is easier to overwrite that former method. Should I revert for now?

Reply:
It would be good to keep the PR within its scope, and not include this. If/when it's needed it can go in.